docs/cvl/cvl2/changes.md

```diff
@@ -0,0 +1,721 @@
+`_.f` {ref}`syntax <cvl2-wildcards>`) must include a `returns` clause if the
+contract method returns a value.  A specific-contract entry may only omit the
+`returns` clause if the contract method does not return a value.
+
+The Prover will report an error if the contract method's return type differs
+from the type declared in the `methods` block entry.
+
+```{todo}
+Error message
+```
+
+Wildcard entries must not declare return types, because they may apply to
+multiple methods that return different types.
+
+```{todo}
+Error message
+```
+
+If a wildcard entry has a ghost or function summary, the user must explicitly
+provide an `expect` clause to the summary.  The `expect` clause tells the
+Prover how to encode the value returned by the summary.  For example:
+
+```cvl
+methods {
+    function _.foo() internal => fooImpl() expect uint256 ALL;
+}
+```
+
+This entry will replace any call to any internal function `f()` with a call to
+the CVL function `fooImpl()`, and will encode the output of `fooImpl` as a
+`uint256`.
+
+If a function does not return any value, the summary should be declared with
+`expect void`.
+
+```{todo}
+Error message
+```
+
+````{warning}
+The Prover is unable to check that the return type declared in the `expect`
+clause matches the return type that the contract expects.  Continuing the above
+example, suppose the contract being verified declared a method `foo()` that
+returns a type other than `uint256`:
+
+```solidity
+function foo() internal returns(address) {
+    ...
+}
+
+function bar() internal {
+    address x = foo();
+}
+```
+
+In this case, the Prover would encode the value returned by `fooImpl()` as a
+`uint256`, and the `bar` method would then attempt to decode this value as an
+`address`.  This will cause undefined behavior, and in some cases the Prover
+will not be able to detect the error.
+````
+
+Changes to integer types
+------------------------
+
+In CVL 1, the rules for casting between integer types were complex; CVL 2
+simplifies them.
+
+The general rule of thumb is that you should use `mathint` whenever possible;
+only use `uint` or `int` types for data that will be passed as input to
+contract functions.
+
+It is now impossible for CVL math expressions to cause overflow - all integer
+operations are exact.
+The remainder of this section describes the changes in detail.
+
+(cvl2-mathops-return-mathint)=
+### Mathematical operations return `mathint`
+
+In CVL 2, the results of all arithmetic operators have type `mathint`,
+regardless of the input types.  Arithmetic operators include `+`,
+`*`, `-`, `/`, `^`, and `%`, but not bitwise operators like `<<`, `xor`, and `|`
+(changes to bitwise operators are described {ref}`below <cvl2-bitwise>`).
+
+The primary impact of this change is that you may need to declare more of your
+variables as `mathint` instead of `uint`.  If you are performing arithmetic
+operations and integers that you are passing to specs, you will need to be more
```

Comment:
I am having trouble parsing this sentence as written. Is it possible that we want to say "...performing arithmetic operations and _generating_ integers..." or something equivalent?